nncf/common/accuracy_aware_training/training_loop.py

```diff
@@ -243,8 +243,8 @@ def run(self, model, train_epoch_fn, validate_fn,
                                                    self.runner.cumulative_epoch_count)
 
             self.runner.train_epoch(model, self.adaptive_controller)
-            self.runner.dump_statistics(model, self.adaptive_controller)
             compressed_model_accuracy = self.runner.validate(model)
+            self.runner.dump_statistics(model, self.adaptive_controller)
             self.runner.accuracy_bugdet = compressed_model_accuracy - self.runner.minimal_tolerable_accuracy
```

Comment:
`dump_statistics()` requires the latest validation results to save the checkpoint - that is why `dump_statistics()` should go after `validate()` I guess

Reply:
It doesn't matter here. Because validation metrics are dumped in `validate()`, which is possible is not the right solution. Anyway, I suggest don't make this change

Reply:
After discussion, it looks that these changes are necessary and fix a bug